openwisp_utils/qa.py

```diff
@@ -51,3 +53,166 @@ def check_migration_name():
+        if len(args.message.split("\n")) > 1:
+            api_message = args.message.split("\n")[1:]
+    else:
+        api_title = commit['title']
+
+        if commit['message'] != '':
+            api_message = commit['message'].split("\n")
+
+    print(api_title)
+    # api_repo = args.repo
+    errors = []
+    # Check dot in end of the first line
+    if api_title[len(api_title.strip()) - 1].strip() == '.':
+        errors.append("Do not add a final dot at the end of the first line.")
+
+    # Check prefix
+    print("Commit Title " + api_title)
+    prefix = re.match(r'\[(.*?)\]', api_title)
+    print(prefix)
+    if not prefix:
+        errors.append(
+            "Add prefix in beginning of the commit. "
+            "Example: [module/file/feature]"
+        )
+    else:
+        # Check capital after prefix
+        commitMessagefirst = api_title.replace(prefix.group(), '').strip()
+        if not commitMessagefirst[0].isupper():
+            errors.append("No capital letter after prefix")
+
+    # Check hastag
+    hastag = re.search(r'\#(\d+)', api_title)
+    print(hastag)
+    if not hastag:
+        errors.append("No mention issues in the short description")
+
+    # Check blank page before first line and second line
+
+    print('api_message' in locals())
+    if 'api_message' in locals():
+        if len(api_message) > 1:
+            if api_message[1] != '':
+                errors.append("No blank line before first line and second line")
+
+    # Check is error
+    if len(errors) == 0:
+        body = "All check done, no errors."
+    else:
+        body = "You have errors with commit message: \n"
+        for e in errors:
+            body += "- " + e + "\n"
+
+    # Without comment to repo
+    # search = requests.get(
+    #     api +
+    #     '/search/issues?q=typetype:pr%20sha:' +
+    #     _mb_substr(commit['hash'], 0, 7) +
+    #     '%20is_unmerged' +
+    #     '&&sort=updated'
+    # )
+    # search_body = search.json()
+    # if len(search_body['items']) > 0:
+    #     search_data = search_body['items'][0]['number']
+    #     response_url = (
+    #         "/repos/" + str(api_repo) +
+    #         "/issues/" + str(search_data) + "/comments")
+    #     headers = {
+    #         'Authorization': 'token %s' % api_token
+    #     }
+    #     post_fields = {
+    #         'body': body
+    #     }
+    #     requests.post(
+    #         api +
+    #         response_url,
+    #         json=post_fields,
+    #         headers=headers)
+    if(len(errors) > 0):
+        raise Exception(body)
+    else:
+        print(body)
+
+
+# def _mb_substr(s, start, length=None):
+#     return (s[start:(start + length)])
+
+
+def _shell_exec(cmd):
+    return os.popen(cmd).read()
+
+
+# Thank you https://gist.github.com/simonw/091b765a071d1558464371042db3b959
+def _get_commit():
+    leading_4_spaces = re.compile('^    ')
+    lines = _shell_exec('git log -1').split('\n')
```

Comment:
I don't think you need to do this, we'll pass output of git log via the bash shell